### PR TITLE
Simplify test job tracker v2.py

### PR DESCRIPTION
**Pull request by bossie**
Adds test for https://github.com/eu-cdse/openeo-cdse-infra/issues/35.

### CODE REVIEW

File: tests/test_job_tracker_v2.py [hunk 1]
```diff
@@ -356,6 +316,20 @@ def submit(
         return app
 
 
+class PrometheusMock:
```
Comment (bossie) [inline tests/test_job_tracker_v2.py:319]: Replace with `mock.Mock(Prometheus)`.

See `tests.test_job_tracker_v2.TestK8sStatusGetter.test_cpu_and_memory_usage_not_in_prometheus`.

**Comment by bossie** [comment]
A bit of context: adding a test for the `K8sStatusGetter`, where the Prometheus API has no records for the CPU/memory usage of a batch job, in which case:

- a warning is issued so we are aware of this situation,
- default values for CPU/memory are reported instead.

To simulate Prometheus returning no values for CPU/memory usage, I found adapting `KubernetesMock` complicated so I simplified things a bit. It essentially comes down to avoid patching module level functions like `openeogeotrellis.job_tracker_v2.kube_client` and `requests.get` and providing mock objects instead; it's still mocking but on a higher level.

Passing `kube_client` and `Prometheus` dependencies explicitly to `K8sStatusGetter` instead of having them buried deeply in the implementation is also a plus:

https://github.com/Open-EO/openeo-geopyspark-driver/blob/f4887244ab78da0b7b16b3bc45ffbb1512681969/openeogeotrellis/job_tracker_v2.py#L594-L595

**Comment by bossie** [comment]
Mostly test code, easy to revert so merged.

**Comment by soxofaan** [comment]
yeah sorry, I had this on my todo list for a long time but never got to it  :smirk:

I also did some tweaking to the job tracker v2 utilities in the meantime to support #531  FYI

**Pull request merged**